src/main/java/Student.java

```diff
@@ -0,0 +1,52 @@
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Student class contains students name and data about the course, start date of the course etc.
+ */
+public class Student {
```

Comment:
You might check the `lombok` library, it allows to cleanup by reducing the amount of boiler plate code. You might want to use `@Data` or at least `@Getter` and `@Setter` provided by `lombok`.